simd/src/Kokkos_SIMD.hpp

```diff
@@ -139,11 +147,15 @@ class abi_set {};
 template <typename... Ts>
 class data_types {};
 
-#if defined(KOKKOS_ARCH_AVX512XEON)
+// FIXME_NVHPC nvc++ doesn't seem to support AVX512.
+#if !defined(KOKKOS_COMPILER_NVHPC) &&  \
+    (defined(KOKKOS_ARCH_AVX512XEON) || \
+     (defined(KOKKOS_ENABLE_IMPL_SIMD_NATIVE) && defined(__AVX512F__)))
 using host_abi_set  = abi_set<simd_abi::scalar, simd_abi::avx512_fixed_size<8>>;
 using data_type_set = data_types<std::int32_t, std::uint32_t, std::int64_t,
                                  std::uint64_t, double>;
-#elif defined(KOKKOS_ARCH_AVX2)
+#elif defined(KOKKOS_ARCH_AVX2) || \
+    (defined(KOKKOS_ENABLE_IMPL_SIMD_NATIVE) && defined(__AVX2__))
 using host_abi_set = abi_set<simd_abi::scalar, simd_abi::avx2_fixed_size<4>>;
 using data_type_set =
     data_types<std::int32_t, std::int64_t, std::uint64_t, double>;
```

Comment:
Why  does `KOKKOS_ENABLE_IMPL_SIMD_NATIVE` not work for NEON> (line 162)?

Reply:
We are already auto-detecting the architecture for `NEON`. We could discuss if we would hide that behind `KOKKOS_ENABLE_IMPL_SIMD_NATIVE` or not given that we don't have an explicit architecture option for `NEON`.

The focus at the moment is on all the changes not related to the new option and I will split the pull request/remove the changes adding testing before merging so that we can review the proposed option separately.